internal/runner/runner.go

```diff
@@ -140,9 +143,21 @@ func (r *CommandRunner) runStep(ctx context.Context, step *CommandStep, dry bool
 	}
 
 	return &CommandStep{
-		Command: append([]string{name}, args...),
-		// NOTE: We don't actually include POSIX env vars as they are redundant.
-		Env:        env,
+		Command:    append([]string{name}, args...),
+		Env:        userEnv,
 		WorkingDir: pwd,
 	}, nil
 }
+
+func dedupEnv(env []string) []string {
+	var deduped []string
+	seen := map[string]bool{}
+	for i := len(env) - 1; i >= 0; i-- {
+		k, _, found := strings.Cut(env[i], "=")
+		if found && !seen[k] {
+			deduped = append(deduped, env[i])
```

Comment:
This results in a slice that is in reverse of the order of input. We should preserve the ordering.